src/main/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixtures/mainIdeWindow/toolWindowsPane/ToolWindowsPane.java

```diff
@@ -42,211 +36,117 @@
+     * Open gradle build tool pane
      *
-     * @return Project View tree fixture
+     * @return the Gradle Build Tool Pane fixture
      */
-    public JTreeFixture projectViewTree() {
-        return find(JTreeFixture.class, JTreeFixture.Companion.byType(), Duration.ofSeconds(10));
+    public GradleBuildToolPane openGradleBuildToolPane() {
+        return (GradleBuildToolPane) togglePane(ButtonLabels.gradleStripeButtonLabel, GradleBuildToolPane.class, true);
     }
 
     /**
-     * Get the Maven Tab tree fixture
-     *
-     * @return Maven Tab tree fixture
+     * Close gradle build tool pane
      */
-    public JTreeFixture mavenTabTree() {
-        return find(JTreeFixture.class, byXpath("//div[@class='SimpleTree']"));
+    public void closeGradleBuildToolPane() {
+        togglePane(ButtonLabels.gradleStripeButtonLabel, GradleBuildToolPane.class, false);
     }
 
     /**
-     * Get the Gradle Tab tree fixture
+     * Create fixture for the Stripe button
      *
-     * @return Gradle Tab tree fixture
+     * @param label label text of the stripe button
+     * @return fixture for the Stripe button
      */
-    public JTreeFixture gradleTabTree() {
-        return find(JTreeFixture.class, byXpath("//div[@class='ExternalProjectTree']"));
+    public JButtonFixture stripeButton(String label) {
+        return stripeButton(label, false);
     }
 
     /**
-     * Get the Build Status tree fixture
+     * Create fixture for the Stripe button
      *
-     * @return Build Status tree fixture
-     */
-    private JTreeFixture buildStatusTree() {
-        return find(JTreeFixture.class, byXpath("//div[@class='Tree']"));
-    }
-
-    /**
-     * Enumeration with building tools
+     * @param label        label text of the stripe button
+     * @param isPaneOpened true if the pane is already opened
+     * @return fixture for the Stripe button
      */
-    public enum ToolToBuildProject {
-        MAVEN("Maven"),
-        GRADLE("Gradle");
-
-        private final String textRepresentation;
-
-        ToolToBuildProject(String textRepresentation) {
-            this.textRepresentation = textRepresentation;
+    public JButtonFixture stripeButton(String label, boolean isPaneOpened) {
+        if (isPaneOpened) {
+            if (label.equals(ButtonLabels.mavenStripeButtonLabel) || label.equals(ButtonLabels.gradleStripeButtonLabel)) {
+                return button(byXpath("//div[@disabledicon='toolWindow" + label + ".svg']"), Duration.ofSeconds(2));
+            } else if (label.equals(ButtonLabels.projectStripeButtonLabel)) {
+                return button(byXpath("//div[@tooltiptext='Project']"), Duration.ofSeconds(2));
+            }
         }
+        return button(byXpath("//div[@text='" + label + "']"), Duration.ofSeconds(2));
+    }
 
-        @Override
-        public String toString() {
-            return textRepresentation;
+    private CommonContainerFixture togglePane(String label, Class fixtureClass, boolean openPane) {
```

Comment:
how about generics instead of type casts

Reply:
Fixed.